docs/capabilities.md

```diff
@@ -0,0 +1,78 @@
+# Capabilities of an user required for a ceph-csi in an Ceph cluster
+
+Ceph uses the term capabilities to describe authorizing an authenticated user
+to exercise the functionality of the monitors, OSDs and metadata servers.
+Capabilities can also restrict access to data within a pool or pool namespace.
+A Ceph administrative user sets a user's capabilities when creating or
+updating a user.
+
+## RBD
+
+We have provisioner, controller expand and node stage secrets in storage class.
+For the provisioner and controller expand stage secret in storageclass, the
+user needs to have the  below mentioned ceph capabilities.
+
+```text
+"mon", "profile rbd",
+"mgr", "allow rw",
+"osd", "profile rbd"
+```
+
+And for the node stage secret in storageclass, the user needs to have the
+below mentioned ceph capabilities.
+
+```text
+"mon", "profile rbd",
+"osd", "profile rbd",
+"mgr", "allow rw"
+```
+
+## CephFS
+
+Similarly in CephFS, for the provisioner and controller expand stage secret in
+storageclass, the user needs to have the below mentioned ceph capabilities.
+
+```text
+"mon", "allow r",
+"mgr", "allow rw",
+"osd", "allow rw tag cephfs metadata=*"
+```
+
+And for node stage secret in storageclass, the user needs to have
+the below mentioned ceph capabilities.
+
+```text
+"mon", "allow r",
+"mgr", "allow rw",
+"osd", "allow rw tag cephfs *=*",
+"mds", "allow rw"
+```
+
+To get more insights on capabilities of cephfs you can refer
+[this document](https://ceph.readthedocs.io/en/latest/cephfs/client-auth/)
+
+## Command to a create user with required capabilities
```

Comment:
the "create user" should be explained a little in this (non-existing) paragraph. Below the user is `kubernetes` (the part after `client.`), and the username is configured in the StorageClass too.
We might even have two users in the StorageClass? One admin (provisioner) and one client (node)?

Reply:
these are the users configured in secrets, not storage-class. yes we have two users one for provisioning and one for mapping/mounting

Reply:
@Madhu-1 and @nixpanic so, shall I explain more about the create user here?

Reply:
you need to explain about the `kubernetes` user as `user` need to be replaced in secret also if you are planning to use a different user other than `kubernetes`

Reply:
I have mentioned it, as the user is subjected to change as per the requirements.